### PR TITLE
feat: use https registry

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

**Review by phoban01 — APPROVED** [review]

File: shared/oci.go [hunk 1]
```diff
@@ -108,8 +108,8 @@ type ComponentModification func(compvers ocm.ComponentVersionAccess) error
 
 // AddComponentVersionToRepository takes a component description and optional resources. Then pushes that component
 // into the locally forwarded registry.
-func AddComponentVersionToRepository(component Component, repository string, componentModifications ...ComponentModification) error {
-	baseURL := "http://127.0.0.1:5000/" + repository
+func AddComponentVersionToRepository(component Component, repository, scheme string, componentModifications ...ComponentModification) error {
+	baseURL := scheme + "://127.0.0.1:5000/" + repository
```
Comment (souleb) [inline shared/oci.go:112]: Can you add a validation step that would parse the `baseURL`?

Comment (phoban01) [reply shared/oci.go:112]: +1

Comment (Skarlso) [reply shared/oci.go:112]: 👍

Comment (Skarlso) [reply shared/oci.go]: Added a parse. But I'm not checking if the scheme is actually a known scheme. That would be overkill here I believe? :)

**Review by souleb — APPROVED** [review]

**Pull request merged**